subsys/trusted_storage/Kconfig

```diff
@@ -0,0 +1,215 @@
+# Subsystem configuration options
+
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig TRUSTED_STORAGE
+	bool "Trusted Storage"
+	help
+	  The trusted storage subsystem allows its users to store in a trusted
+	  way device intimate data and sensitive intimate secrets, to ensure
+	  data privacy or data integrity.
+	  It provides multiple APIs with different trust goals.
+	  It supports several trusted implementation back-ends to provide
+	  various levels of trust depending on the device security features.
+
+if TRUSTED_STORAGE
+module = TRUSTED_STORAGE
+module-str = trusted_storage
+source "subsys/logging/Kconfig.template.log_config"
+endif
+
+config TRUSTED_STORAGE_SETTINGS_HELPERS
+	bool "Trusted Storage Settings Helpers"
+	depends on SETTINGS
+	depends on SETTINGS_RUNTIME
+	help
+	  Build Settings subsystem based Trusted Storage helpers.
+
+config TRUSTED_STORAGE_SETTINGS_AUTH_CRYPT_BACKEND
+	bool "Trusted Storage Authenticated Encryption Backend"
+	select TRUSTED_STORAGE_SETTINGS_HELPERS
+	help
+	  Build Trusted Storage Authenticated Encryption Backend.
+
+config TRUSTED_STORAGE_AUTH_CRYPT_BACKEND_PSA
+	bool "Trusted Storage Authenticated Encryption with PSA Crypto"
+	depends on MBEDTLS_BUILTIN_PSA_CRYPTO
+	depends on MBEDTLS_CIPHER
```

Comment:
These should be converted to PSA_WANT_ALG_XXXX, but I think you need a PR from me in Zephyr before you can open up a pure PR upstream.

The PSA_WANT_ALG_XXXX is suported in our sdk-zephyr